src/shogun/mathematics/linalg/LinalgBackendViennacl.h

```diff
@@ -261,6 +272,18 @@ class LinalgBackendViennaCL : public LinalgBackendGPUBase
 		return viennacl::linalg::inner_prod(a_gpu->data_vector(a.size()), b_gpu->data_vector(b.size()));
 	}
 
+	/** ViennaCL matrix in-place elementwise product method */
+	template <typename T>
+	void element_prod_impl(SGMatrix<T>& a, SGMatrix<T>& b, SGMatrix<T>& result) const
+	{
+		GPUMemoryViennaCL<T>* a_gpu = cast_to_viennacl(a);
+		GPUMemoryViennaCL<T>* b_gpu = cast_to_viennacl(b);
+		GPUMemoryViennaCL<T>* result_gpu = cast_to_viennacl(result);
+
+		result_gpu->data_matrix(a.num_rows, a.num_cols) = viennacl::linalg::element_prod(
```

Comment:
btw do you have a gpu so that this code is actually tested on your machine?
Remember that our new implementation falls back to eigen3 if it is not available .....